src/kaleidoscope/hardware/technomancy/Atreus2.cpp

```diff
@@ -25,14 +25,12 @@ namespace kaleidoscope {
 namespace hardware {
 namespace technomancy {
 
-ATMEGA_KEYBOARD_DATA(Atreus2);
-constexpr int8_t Atreus2::led_count;
+ATMEGA_KEYSCANNER_IMPLEMENTATION();
 
 }
 }
 }
 
 HARDWARE_IMPLEMENTATION KeyboardHardware;
```

Comment:
It feels a little weird to have this boilerplate code here that's a one-liner that uses a one-line macro that's defined in the .h file.  (Also, the new code expands to

kaleidoscope::hardware::technomancy::Atreus2 KeyboardHardware;

while the old code was 

kaleidoscope::hardware::technomancy::Atreus2 &Atreus2 = KeyboardHardware;	

Is that right?

Reply:
My greps were insufficiently wide. I do see that we're using HARDWARE_IMPLEMENTATION elsewhere in the codebase.

I do wonder if we want to be calling this something other than HARDWARE_IMPLEMENTATION and KeyboardHardware.

I don't have good names at the moment, though.

Reply:
The old code had _both_, with the latter being a "friendlier" name for the former. But only some boards had that, not all of them.

Sadly, we need _something_ in here, to implement all the derived data structures (iow, all the static data in the keyscanner and its props). We could do that with more macros, but none of those would be useful in any other context, which is why I opted for a single big one.

The namespace surroundings are there in case any board needs to add further functionality to its device, or any of the components, that are outside of the parts that `_IMPLEMENTATION()` covers.